internal/controller/pkg/revision/hook.go

```diff
@@ -154,6 +159,9 @@ func (h *ProviderHooks) Post(ctx context.Context, pkg runtime.Object, pr v1.Pack
 		return err
 	}
 	s, d, svc, secSer, secCli := buildProviderDeployment(pkgProvider, pr, cc, h.namespace, append(pr.GetPackagePullSecrets(), ps...))
+	if secSer == nil || secCli == nil {
+		return errors.New(errNoTLSSecretNameAssigned)
+	}
```

Comment:
here it makes sense to error out, as it means something went wrong and we would otherwise end up with a broken installation.